script/input/1/config.json

```diff
@@ -0,0 +1,21 @@
+{
+    "chainlog": "0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F",
+    "admin": "0xBE8E3e3618f7474F8cB1d074A26afFef007E98FB",
+    "hub": "0x12F36cdEA3A28C35aC8C6Cc71D9265c17C74A27F",
+    "type": "aave",
+    "ilk": "DIRECT-AAVEV2-DAI",
+    "maxLine": 5000000,
+    "gap": 3000000,
+    "ttl": 28800,
+    "tau": 604800,
+    "aave": {
```

Comment:
I'm a bit confused about this structure of file. The `ilk` and auto-line settings are per each D3M. Why are they in an upper level and then we have `aave` and `compound` sub values?

Reply:
I guess we define the general and then the specifics, but I feel a bit confusing to have it mixed this way. As I'm gonna a deploy aave (for example) my config file won't have anything about compound.

Reply:
I'm open to alternatives, but I think this makes sense as both Aave and Compound adapters share all the parameters in the root object. They both have debt ceilings and whatnot.

It's possible you are thinking this are parameters that should be defined per-d3m? These are just sensible defaults to get up and running quickly and most real deploys will override whatever they need to via environment variables. There are examples of this in the OP examples above.

In the case of the spell init this will be hard coded in the spells same as always and these values will be unused.

Reply:
I think it might be cleaner to try and use only the config file parameters and not a mixup of env vars and the config file parameters.
That way there is less chance for mistakes (it's always confusing to use overriding env vars in a script). Also this way the config file is an afterward record of what was used (assuming the changes are committed).

Reply:
We could use a full json config file as an override, but the structure of having a file-based override is not that clean. There are two points:

1. There may be more than just "official" Aave and Compound deploys. People can deploy forks of these protocols. If/when we add something like Centrifuge D3Ms they will have multiple deploys as well. (5 instances are live now). It's a bit awkward to have the configs here match what's live. Do we keep these synced when parameters changes are made over time after initial deploy/init?
2. If I want to deploy a D3M to a chain fork which is the sole use-case for even calling `init` with a config file then it is much easier to fork the repo as-is and just define the environment variable overrides. Otherwise you need to either add a json file config override to the repository or add file permissions to some random file elsewhere in the system. A common pattern for this will be to have a higher-level deploy script repository which imports `dss-direct-deposit` and other dependencies.

An example structure:

```
dependencies/dss-teleport/...
dependencies/xdomain-dss/...
dependencies/dss-bridge/...
deploy-domain.sh
init-domain.sh
```

`deploy.sh` and `init.sh` would then just be able to change into the dependencies directories of each repository and run each deploy/init script respectively against an anvil fork without having to modify the repositories. As I said feeding in an override config is possible too, but we have to deal with file permissions then.

Overall I would prefer a higher level coordination structure defined in Foundry, and there has been an [issue opened for this](https://github.com/foundry-rs/foundry/issues/3911), but for now the environment variable overrides works quite nicely, but again if you guys feel strongly about going the override route we can find something that works.

Reply:
Foundry team seems like they are willing to move on this actually. I've added a point about standardizing global config with the `deploy` command: https://github.com/foundry-rs/foundry/issues/3911#issuecomment-1369962540

Let's hold off merging this until we see what that looks like.

Reply:
Honestly my point was more about that looks a bit weird the example file has defined an `ilk` called `DIRECT-AAVEV2-DAI` and it says type `aave` and then we have a compound section.

So I would expect something more like:
```
{
    "chainlog": "0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F",
    "admin": "0xBE8E3e3618f7474F8cB1d074A26afFef007E98FB",
    "hub": "0x12F36cdEA3A28C35aC8C6Cc71D9265c17C74A27F",
    "type": "aave",
    "ilk": "DIRECT-AAVEV2-DAI",
    "maxLine": 5000000,
    "gap": 3000000,
    "ttl": 28800,
    "tau": 604800,
    "lendingPool": "0x7d2768dE32b0b80b7a3454c06BdAc94A69DDc7A9",
    "king": "0xBE8E3e3618f7474F8cB1d074A26afFef007E98FB",
    "bar": 370
}
```
and
```
{
    "chainlog": "0xdA0Ab1e0017DEbCd72Be8599041a2aa3bA7e740F",
    "admin": "0xBE8E3e3618f7474F8cB1d074A26afFef007E98FB",
    "hub": "0x12F36cdEA3A28C35aC8C6Cc71D9265c17C74A27F",
    "type": "compound",
    "ilk": "DIRECT-COMP-DAI",
    "maxLine": 5000000,
    "gap": 3000000,
    "ttl": 28800,
    "tau": 604800,
    "cdai": "0x5d3a536E4D6DbD6114cc1Ead35777bAB948E3643",
    "king": "0xBE8E3e3618f7474F8cB1d074A26afFef007E98FB",
    "barb": 7535450719
}
```

Reply:
I can agree with Tal, that replacing with env variables is confusing.
IMO what we should leave in the repo is a config example and then have to pass this config file as a flag of the deployment scripts. So user locally defines the file and then pases the path to it when executing.

Reply:
Ah I see what you're saying. Ok yes I like that structure better. Can split the config file into two examples (Aave and Compound) and pass in full config via cli somehow (it was awkward to do this as you can't read cli args AFAIK).

Reply:
I guess we use the bash script to get the file from the flag and pass the values. At least in the meanwhile we use the bash.

Reply:
Updated to use only json config files.